NadekoBot.Core/Modules/Searches/SearchesSlashCommands.cs

```diff
@@ -0,0 +1,119 @@
+using AngleSharp;
+using AngleSharp.Dom;
+using AngleSharp.Html.Dom;
+using AngleSharp.Html.Parser;
+using Discord;
+using Discord.Interactions;
+using Discord.WebSocket;
+using Microsoft.Extensions.Caching.Memory;
+using NadekoBot.Common;
+using NadekoBot.Common.Attributes;
+using NadekoBot.Common.Replacements;
+using NadekoBot.Core.Modules.Searches.Common;
+using NadekoBot.Core.Services;
+using System.Collections.Generic;
+using NadekoBot.Extensions;
+using NadekoBot.Modules.Searches.Common;
+using NadekoBot.Modules.Searches.Services;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Drawing.Processing;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using NadekoBot.Modules.Administration.Services;
+using Serilog;
+using Configuration = AngleSharp.Configuration;
+
+namespace NadekoBot.Modules.Searches
+{
+    public partial class SearchesSlash : NadekoSlashModule<SearchesService>
+    {
+        private readonly IHttpClientFactory _httpFactory;
+
+        private readonly GuildTimezoneService _tzSvc;
+        public SearchesSlash(IHttpClientFactory factory,GuildTimezoneService tzSvc)
+        {
+            _httpFactory = factory;
+            _tzSvc = tzSvc;
+        }
+
+        [NadekoSlash]
+        public async Task Weather([Summary("city","name of the city")]string city,
+        [Summary("country","optional name of country")]string country = "")
+        {
+            if (!await ValidateQuery(ctx.Channel, city).ConfigureAwait(false))
+                return;
+            string query;
+            await ctx.Interaction.DeferAsync().ConfigureAwait(false);
+            // case for no country
+            if (country == ""){
+                query = city;
+            //lookup country code using json file
+            }else{
+                using (var _http = _httpFactory.CreateClient()){
+                    string strdata = await _http.GetStringAsync("https://datahub.io/core/country-list/r/data.json");
+                    List<CountryCode> dd = JsonConvert.DeserializeObject<List<CountryCode>>(strdata);
+                    string code = dd.Where(x=>x.Name.ToLowerInvariant()==country.ToLowerInvariant()).Select(e => e.Code).FirstOrDefault().ToString();
+                    if (code == null){
+                        query = city; 
+                    }else{
+                        query = $"{city},{code}";
+                    }
+                }
+            }
```

Comment:
A sketch of a clearer pattern for doing this might be
```cs
// Query just contains a city by default
string query = city;

// Try to look up a country code and add it to the query
if (country != "") {
    // do stuff
  string code = ...
  if (code != null) {
    query = ...
  }
}
```

Reply:
the code is changed to use this pattern